pycoast/cw_base.py

```diff
@@ -924,24 +947,131 @@ def add_cities(self, image, area_def, citylist, font_file, font_size,
                         self._draw_ellipse(
                             draw, dot_box, fill=outline, outline=outline)
-                        text_position = [x + 9, y - 5]  # FIX ME
+                        text_position = [x + 9, y - 5]  # FIXME
                     else:
                         text_position = [x, y]
 
                 # add text_box
-                    self._draw_text_box(draw, text_position, city_name, font,
-                                        outline, box_outline, box_opacity)
+                    self._draw_text_box(draw, text_position, city_name, font, outline,
+                                        box_outline, box_opacity)
                     logger.info("%s added", str(city_name))
 
         self._finalize(draw)
 
+    def add_points(self, image, area_def, points_list, font_file, font_size=12,
+                   symbol='circle', ptsize=6, outline='black', fill='white', **kwargs):
+        """Add a symbol and/or text at the point(s) of interest to a PIL image object.
+        :Parameters:
+            image : object
+                PIL image object
+            area_def : object
+                Area Definition of the provided image
+            points_list : list [((lon, lat), desc)]
+              | a list of points defined with (lon, lat) in float and a desc in string
+              | [((lon1, lat1), desc1), ((lon2, lat2), desc2)]
+              | lon : float
+              |    longitude of a point
+              | lat : float
+              |    latitude of a point
+              | desc : str
+              |    description of a point
+            font_file : str
+                Path to font file
+            font_size : int
+                Size of font
+            symbol : string
+                type of symbol, one of the elelment from the list
+                ['circle', 'square', 'asterisk']
+            ptsize : int
+                Size of the point.
+            outline : str or (R, G, B), optional
+                Line color of the symbol
+            fill : str or (R, G, B), optional
+                Filling color of the symbol
+
+        :Optional keyword arguments:
+            width : float
+                Line width of the symbol
+            outline_opacity : int, optional {0; 255}
+                Opacity of the line of the symbol.
+            fill_opacity : int, optional {0; 255}
+                Opacity of the filling of the symbol
+            textbox_outline : str or (R, G, B), optional
+                Line color of the textbox borders.
+            textbox_linewidth : float
+                Line width of the the borders of the textbox
+            textbox_fill : str or (R, G, B), optional
+                Filling color of the background of the textbox.
+            textbox_opacity : int, optional {0; 255}
+                Opacity of the background filling of the textbox.
+        """
+        from pyresample.geometry import AreaDefinition
```

Comment:
Pyresample is not technically a dependency of pycoast. How do other methods handle this type of logic? I know the lon/lat -> x/y coordinates could be done with only Proj, but not to x/y grid indexes.

I'm ok with pyresample being depended on here, but would prefer there by a try/except ImportError and a message about points requiring pyresample.

Reply:
The reason I add the import and the isinstance() check is because other functions, except add_cities(), support two types of area_def, like

> area_def : (proj_str, area_extent) or AreaDefinition

However, add_cities() and add_points() only support area_def parameter as AreaDefinition object. In add_cities() there is a try/except when use get_xy_from_lonlat(), but I think "Explicit is better than implicit", so I add the type check at the beggining of the function, which needs a explicit import of pyresample. I will add a try/except to that.